http/src/main/java/io/micronaut/http/cookie/DefaultServerCookieEncoder.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.cookie;
+
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.util.StringUtils;
+
+import java.net.HttpCookie;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Implementation of {@link ServerCookieEncoder} which uses {@link HttpCookie} to encode cookies.
+ * @author Sergio del Amo
+ * @since 4.4.0
+ */
+@Internal
+public class DefaultServerCookieEncoder implements ServerCookieEncoder {
```

Comment:
can this be `final`? Does it need to be `public`?

Reply:
> can this be final? 

https://github.com/micronaut-projects/micronaut-core/pull/10402/commits/a8d3204d9c512fcc50950358fd14bf25905b6bbe

> Does it need to be public?

Yes, it needs to be public.